src/mempool_prioritization.cpp

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2019-2020 Xenios SEZC
+// https://www.veriblock.org
+// Distributed under the MIT software license, see the accompanying
+// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
+
+#include "veriblock/mempool_prioritization.hpp"
+
+namespace altintegration {
+
+bool isStronglyEquivalent(const VTB& vtb1,
+                          const VTB& vtb2,
+                          VbkBlockTree& tree) {
+  auto* vtb1_containing_block =
+      tree.getBlockIndex(vtb1.containingBlock.getHash());
+  auto* vtb2_containing_block =
+      tree.getBlockIndex(vtb2.containingBlock.getHash());
+
+  VBK_ASSERT_MSG(vtb1_containing_block,
+                 "vtb1 unknown containing block %s",
+                 vtb1.containingBlock.toPrettyString());
+  VBK_ASSERT_MSG(vtb2_containing_block,
+                 "vtb1 unknown containing block %s",
+                 vtb2.containingBlock.toPrettyString());
+
+  bool is_on_the_same_chain = false;
+  if (vtb1_containing_block->getHeight() > vtb2_containing_block->getHeight()) {
+    is_on_the_same_chain =
+        vtb1_containing_block->getAncestor(
+            vtb2_containing_block->getHeight()) == vtb2_containing_block;
+  } else {
+    is_on_the_same_chain =
+        vtb2_containing_block->getAncestor(
+            vtb1_containing_block->getHeight()) == vtb1_containing_block;
+  }
+
+  return (vtb1.transaction.bitcoinTransaction ==
+          vtb2.transaction.bitcoinTransaction) &&
+         (vtb1.transaction.blockOfProof == vtb1.transaction.blockOfProof) &&
+         is_on_the_same_chain;
```

Comment:
I think whether VTBs are on same chain or not should not imfluence their 'strong equality'
so technically it is better to calculate StronglyEquivalentIds for VTBs (which do not include containing block), and separately add member func to `VbkBlockTree` `bool AreOnSameChain(const VbkBlock& b1, const VbkBlock& b2) const`